src/stores/useSocialMediaStore/useSocialMediaStore.ts

```diff
@@ -2,43 +2,55 @@
 
 import { create } from 'zustand';
 
+import { octopostApi } from '~services/api';
 import { AccountsService } from '~services/api/accounts/accounts';
 import { SocialMediaService } from '~services/api/social-media/social-media';
 import { SocialMedia } from '~services/api/social-media/social-media.types';
-import { MultiMap } from '~utils/multimap/multimap';
+import { GenericObject } from '~types/object';
 
 import { Tab, Tabs } from '~components/Tabber/Tabber.types';
 
 import { SocialMediaState, StoreAccount } from './useSocialMediaStore.types';
 
 export const useSocialMediaStore = create<SocialMediaState>((set) => ({
   accounts: {
-    data: null,
+    data: [],
     error: '',
     loading: false,
   },
 
-  addAccount: async (): Promise<void> => {},
+  addAccount: async (newAccount: StoreAccount): Promise<GenericObject> => {
```

Comment:
como parâmetro, o id também não é obrigatório, sendo assim, o tipo de newAccount não pode ser StoreAccount, ou seja, atribua um novo tipo

Eu recomendaria você utilizar do omit do typescript para tirar apenas o id

`type NewAccount= Omit<StoreAccount, 'id'>;`

```suggestion
  addAccount: async (newAccount: NewAccount): Promise<GenericObject> => {
```